wfsim/core/s2.py

```diff
@@ -139,18 +155,19 @@ def get_electron_yield(n_electron, positions, z_obs, config, resource):
         # Average drift time of the electrons
         drift_time_mean, drift_time_spread = S2.get_s2_drift_time_params(z_obs, positions, config, resource)
 
-        # Absorb electrons during the drift
+        # Absorb electrons during the drift, clipping to avoid artefacts of maps
+        drift_time_mean=np.clip(drift_time_mean, a_min=0, a_max=np.inf)
         electron_lifetime_correction = np.exp(- 1 * drift_time_mean /
                                               config['electron_lifetime_liquid'])
         cy = config['electron_extraction_yield'] * electron_lifetime_correction
 
         # Remove electrons in insensitive volume
         if config['enable_field_dependencies']['survival_probability_map']:
-            survival_probability = resource.field_dependencies_map(z_obs, positions, map_name='survival_probability_map')
-            cy *= survival_probability
-
-        # why are there cy greater than 1? We should check this
-        cy = np.clip(cy, a_min = 0, a_max = 1)
+            p_surv = resource.field_dependencies_map(z_obs, positions, map_name='survival_probability_map')
+            if np.any(p_surv<0) or np.any(p_surv>1):
```

Comment:
Is it necessary this if statement? Clipping simply does nothing if this is false